libsql-sqlite3/src/build.c

```diff
@@ -830,7 +833,7 @@ static void SQLITE_NOINLINE deleteTable(sqlite3 *db, Table *pTable){
   for(pIndex = pTable->pIndex; pIndex; pIndex=pNext){
     pNext = pIndex->pNext;
     assert( pIndex->pSchema==pTable->pSchema
-         || (IsVirtual(pTable) && pIndex->idxType!=SQLITE_IDXTYPE_APPDEF) );
+         || (IsVirtual(pTable) && (pIndex->idxType&3)!=SQLITE_IDXTYPE_APPDEF) ); // '&3' is the LibSQL fix to treat VECTOR index as APPDEF
```

Comment:
I am not sure I understand what this is all about.

In any case, please hide the "&3" thing in a helper function like `getIndexType(pIndex)` to keep it more manageable

Reply:
Added `IsAppDefIndex` helper